packages/auth/amplify_auth_cognito_dart/lib/amplify_auth_cognito_dart.dart

```diff
@@ -28,7 +28,8 @@ export 'src/model/password/cognito_update_password_options.dart';
 export 'src/model/session/cognito_auth_session.dart';
 export 'src/model/session/cognito_auth_user.dart';
 export 'src/model/session/cognito_session_options.dart';
-export 'src/model/session/cognito_sign_in_details.dart';
+export 'src/model/session/cognito_sign_in_details.dart'
+    hide CognitoSignInDetailsFederated;
```

Comment:
Q: why do we want to hide `CognitoSignInDetailsFederated` given it's peers `CognitoSignInDetailsApiBased` and   `CognitoSignInDetailsHostedUi` are not hidden

Reply:
I wanted to leverage the SignInDetails infrastructure for determining `isSignedIn`, but the Auth group wanted the `getUser` API to throw when federated (which is where this information is returned to users). So currently it's just an implementation detail, but I thought it the cleanest way to track this information.

Reply:
understood. so we can also hide `CognitoSignInDetailsApiBased` and `CognitoSignInDetailsHostedUi`.